core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -192,7 +194,7 @@ protected boolean caseSensitive() {
 
   @Override
   public RewriteDataFilesActionResult execute() {
-    CloseableIterable<FileScanTask> fileScanTasks = null;
+    CloseableIterable<FileScanTask> fileScanTasks = CloseableIterable.empty();
```

Comment:
I'm not sure whether this name `RewriteDataFilesActionResult` should be renamed to `RewriteFilesActionResult` because the rewrite action is removing all the deletions from files set, it also rewrite the delete files actually.